### PR TITLE
Refactoring of `prime_as_sum_of_two_squares`

### PR DESCRIPTION
**Pull request by haru-44**
* Added docstring
* We now check if it is prime or not
* When small b, we omitted the calculation of the Legendre symbol

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/solvers/diophantine/diophantine.py [hunk 1]
```diff
@@ -3602,6 +3603,24 @@ def prime_as_sum_of_two_squares(p):
     Represent a prime `p` as a unique sum of two squares; this can
     only be done if the prime is congruent to 1 mod 4.
 
+    Parameters
+    ==========
+
+    p : Integer
+        A prime is congruent to 1 mod 4
```
Comment (smichr) [inline sympy/solvers/diophantine/diophantine.py:3610]:
```suggestion
        A prime that is congruent to 1 mod 4
```

File: sympy/solvers/diophantine/diophantine.py [hunk 2]
```diff
@@ -3614,29 +3633,38 @@ def prime_as_sum_of_two_squares(p):
...
     if p % 8 == 5:
+        # Legendre symbol (2/p) == -1 if p % 8 in [3, 5]
         b = 2
-    else:
+    elif p % 12 == 5:
+        # Legendre symbol (3/p) == -1 if p % 12 in [5, 7]
         b = 3
-
-        while pow(b, (p - 1) // 2, p) == 1:
+    elif p % 5 in [2, 3]:
```
Comment (smichr) [inline sympy/solvers/diophantine/diophantine.py:3656]: OK, I'll bite: why do we explicitly test `in [2,3]` but not do so above, e.g. like `p%8 in [3,5]`?

Comment (smichr) [reply sympy/solvers/diophantine/diophantine.py:3656]: I see - being 1 from 4 means the most you can be from multiples of 4 greater than that is 5.

**Comment by haru-44** [comment]
thanks

**Pull request merged**